documentation/specs/src/interoperability/ethereum-bridge/bootstrapping.md

```diff
@@ -1,117 +1,115 @@
-must remain possible to restart the chain with the Ethereum bridge still not
-enabled.
+To bootstrap the Ethereum bridge, there are six governance parameters which
+must be written to storage:
+
+- `eth_bridge_min_confirmations` - The minimum number of block confirmations
+  on Ethereum required for any given event to be voted on by Namada validators.
+- `eth_bridge_bridge_address` - The address of the `Bridge` contract, used to
+  perform transfers in either direction (Namada <> Ethereum).
+- `eth_bridge_bridge_version` - The version of the `Bridge` contract, starting
+  from 1.
+- `eth_bridge_governance_address` - The address of the `Governance` contract,
+  used to perform administrative tasks, such as updating validator sets in
+  Ethereum.
+- `eth_bridge_governance_version` - The version of the `Governance` contract,
+  starting from 1.
+- `eth_bridge_wnam_address` - The address of the deployment of the native
+  ERC20 token, representing NAM in Ethereum.
+
+Here is an overview of the steps to follow to enable the Ethereum bridge through a
+governance proposal for a given Namada chain:
+
+1. The [smart contracts](./ethereum_smart_contracts.md) of the Ethereum bridge
+   should be deployed a priori, with an initial validator set and other parameters,
+   such as nonces, from epoch $E_0$. This step can be done by anyone, but ultimately
+   it's up to governance validators to decide which addresses to trust.
+2. A governance proposal should be held to agree on an epoch $E_{bridge} > E_0$ at
+   which to launch the Ethereum bridge. The proposal should be executed at some
+   grace epoch $E_{grace}$, with $E_{grace} \le E_{bridge}$. The wasm code of
+   the proposal should modify storage with:
+
+    - The epoch $E_{bridge}$ when the bridge is to be enabled.
+    - The aforementioned governance parameters of the Ethereum bridge.
+
+   Additionally, the knowledge of $E_0$ must be communicated through the governance
+   proposal's JSON data. This is important, as subsequent validator sets from epochs
+   $E$ such that $E > E_0$ should be relayed if the governance proposal passes.
+   The Ethereum block height when the contracts were deployed should also be
+   included in the proposal.
+3. Validators should vote on the proposal if the wasm code correctly updates
+   Namada's storage, the proposal contains the epoch $E_0$ of the first set of
+   validators in the deployed contracts and the Ethereum height at which the
+   contracts were deployed, and the contracts are initialized correctly.
+4. Eventually, the proposal passes at some epoch $E_{end} \le E_{grace}$, if enough
+   validators vote on it. Then, the Ethereum oracle receives an update command, so it
+   can start processing Ethereum blocks to extract confirmed events. Should the proposal
+   not pass, no state updates are applied in Namada, therefore the bridge is not enabled.
+
+Notice that nothing stops users from preemptively updating the smart contracts
```

Comment:
I'm confused here. Do you mean that the person deploying the contract lied about which validator set was in the contract?

Reply:
no, the contract was deployed with the right address, but because the governance proposal is published on-chain as public knowledge, anyone can start pushing validator set updates to the contracts before the governance proposal to enable the bridge passes

Reply:
ah, I see.